### DISCUSSION TITLE
Transition animation

### DISCUSSION THREAD

**Discussion by kuroikana** [Kareadita/Kavita#2843] [Ideas]
### Idea Description

Animation transitions are possible while browsing

### Idea Category

Web UI

### Duration of Using Kavita

1 and a half years

### Before submitting

- [X] I've already searched for existing ideas before posting.

**Comment by therobbiedavis** [comment]
Could you be more specific? 
- What animation were you imagining?
- When browsing where?
- Do you have an example that made you think to request this?

**Comment by Ri-ju** [comment]
I was asking in the discord wether I should create a discussion but a kind soul directed me here, so I would like to contribute to this feature request by adding some specifics.

It would be great to have a slide animation when turning pages in the Kavita web reader (for comics, manga, EPUBs, or PDFs). Currently, pages instantly switch. An optional animation would make reading feel smoother and more natural. It's those little things which excites readers... (Or it's just me. Ha!).

**How it could work:**
Page slides left (or right for Right to Left content) when going forward/backward.
Triggered by swipe, tap, or keyboard navigation.
Fast and smooth slide.
Optional toggle in reader settings for people who prefers without.
Disabled during infinite scrolling contents like webtoon (Unless people want it?)

**App Example:**
Apps like Mihon already have this feature (“Animate page transitions”).

**Reasons for Requesting:**
Enhances reading flow and immersion.
Keeps the reading experience visually continuous.
Users who prefer instant page changes can leave it disabled.

**Notes:**
Perhaps in the future it can be expanded with several transition, like actual paper turning animation if people would prefer that, or slide -> fade out (as the page reaches midscreen roughly during swipe).

However I had some uncertainity about requesting this, as not sure how Kavita works by loading the pages, as swiping back and forth may need previous, current and next page into memory and it might ruin the reading experience, but this is just uneducated thought from non-developer's viewpoint. (Perhaps people with the knowledge would like to chime in with remarks on how this feature could be achieved, in developer-speak!)

Thanks for your time to read this request/contribution!
